flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/ByteBufUtilsTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+import org.apache.flink.shaded.netty4.io.netty.buffer.Unpooled;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+
+/**
+ * Tests the methods in {@link ByteBufUtils}.
+ */
+public class ByteBufUtilsTest {
+
+	@Test
+	public void testAccumulateWithoutCopy() {
+		final int sourceLength = 128;
+		final int sourceStartPosition = 32;
+		final int expectedAccumulationSize = 16;
+
+		ByteBuf src = createSourceBuffer(sourceLength, sourceStartPosition);
+
+		ByteBuf target = Unpooled.buffer(expectedAccumulationSize);
+
+		// If src has enough data and no data has been copied yet, src will be returned without modification.
+		ByteBuf accumulated = ByteBufUtils.accumulate(target, src, expectedAccumulationSize, target.readableBytes());
+
+		assertSame(src, accumulated);
+		assertEquals(sourceStartPosition, src.readerIndex());
+
+		verifyBufferContent(src, sourceStartPosition, sourceLength - sourceStartPosition, sourceStartPosition);
+	}
+
+	@Test
+	public void testAccumulateWithCopy() {
+		final int firstSourceLength = 128;
+		final int firstSourceStartPosition = 32;
+		final int secondSourceLength = 64;
```

Comment:
we can make two sources have same length to achieve the same goal, then to void too many vars.

Reply:
Unified the source length.